### ISSUE TITLE
Unable to outdent #region (and other preprocessor directives)

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#37575]
My team uses a lot of #region directives and many of us, including myself, prefer to have them fully-OUTdented.  This is something that ReSharper specifically allows, but I've stopped using ReSharper since moving to Visual Studio 2019.  However, VS 2019 is now forcing all of my #region statements to be INdented and it's driving me crazy.  Please add a setting that allows us to do this (or even one as fancy as the ReSharper one that allows a number of choices between &quot;indented&quot; and &quot;fully-outdented&quot;...

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/665986/unable-to-outdent-region-and-other-preprocessor-di.html
VSTS ticketId: 957069_
_These are the original issue comments:_

Visual Studio Feedback System on 7/29/2019, 11:22 PM (10 hours ago): <markdown><p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>
</markdown>
_These are the original issue solutions:_
(no solutions)

**Comment by sharwell** [comment]
Related to #2487
Related to #12305
Related to #12306
Mentioned in https://github.com/dotnet/roslyn/pull/37433#discussion_r306530258

**Comment by sharwell** [comment]
🔗 https://developercommunity.visualstudio.com/content/problem/719205/code-style-region-directive-alignment.html

This feedback further requests the ability to select:

* Leave as-is (do not reformat the `#region` or `#endregion` directives themselves)
* Outdent by one

**Comment by hamarb123** [comment]
I no longer use `#region` as a result of this in C# (it still works properly in VB from memory). I would start using them again if I could put them to the left.

**Comment by Gillissie** [comment]
I'm here to add my support for NOT auto-indenting pre-processor directives. There should be an explicit option to make them all non-indented.

**Comment by ObiquitechAdmin** [comment]
Agreed. I use a couple of formatting extensions both with Visual Studio and VS Code, but in their combined functionality there are not yet any features to prevent the pre-processor directives such as `#region`  sections from lining up everything within them. Very annoying.

**Comment by skurth** [comment]
VB wins this round ;-)

**Comment by CyrusNajmabadi** [comment]
Tracking with https://github.com/dotnet/roslyn/issues/31691
